### ISSUE TITLE
print only a slice of an array

### ISSUE THREAD

**Issue by GianniGi** [garu/Data-Printer#116]
Hello, I'm using Data::Printer with arrays, variables and data structures that contain thousands of values, and each value is a very long string, longer than the screen.

Can I suggest to add the option to print only the first elements of an array, including only the first characters for each element? This could make the module really flexible.

Another valuable option would be to print by default the variable or array name along with its data.

Thanks for all the work that you put into this addon.

**Comment by hakonhagland** [comment]
I thinks this might be a good idea, but I will have to leave this up to the maintainers to decide if it is
viable. I haven't looked at the source for some time now, I just wanted to say that I started on the problem
of printing the variable name along with its data 2 years ago, see [this](https://github.com/garu/Data-Printer/pull/74) pull request. ( I am personally using that code in the pull request and it works fine on my computer). There is also a new module [PPR](https://metacpan.org/pod/PPR) that seems promising for parsing Perl and hence allowing the variable name to be printed along with the data.

**Comment by GianniGi** [comment]
Still no news from the mantainers? :-) 
I hope that this improvement will be taken into consideration.

**Comment by garu** [comment]
Sorry for the long hiatus. We're working on it.

Also, many thanks to @hakonhagland for the amazing work and PRs, they are also being looked at for a merge real soon!

**Comment by GianniGi** [comment]
Thanks, very happy to hear this :-)

**Comment by garu** [comment]
Hey @GianniGi and @hakonhagland!

I'm happy to announce the current master branch of Data::Printer (released on CPAN as 0.99_XXX) has added support for this! Now, giant arrays/hashes/strings will be trimmed down to a more reasonable size, like so:

> `perl -MDDP -E 'my $s = join(q(),a..z) x 100; p $s'`
"abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghij(...skipping 1576 chars...)"

and for arrays/hashes:

```
> perl -MDDP -E 'my @a = 33 .. 100; p @a'
[
    [0]  33,
    [1]  34,
    [2]  35,
    [3]  36,
    [4]  37,
    [5]  38,
    [6]  39,
    [7]  40,
    [8]  41,
    [9]  42,
    [10] 43,
    [11] 44,
    [12] 45,
    [13] 46,
    [14] 47,
    [15] 48,
    [16] 49,
    [17] 50,
    [18] 51,
    [19] 52,
    [20] 53,
    [21] 54,
    [22] 55,
    [23] 56,
    [24] 57,
    [25] 58,
    [26] 59,
    [27] 60,
    [28] 61,
    [29] 62,
    [30] 63,
    [31] 64,
    [32] 65,
    [33] 66,
    [34] 67,
    [35] 68,
    [36] 69,
    [37] 70,
    [38] 71,
    [39] 72,
    [40] 73,
    [41] 74,
    [42] 75,
    [43] 76,
    [44] 77,
    [45] 78,
    [46] 79,
    [47] 80,
    [48] 81,
    [49] 82,
         (...skipping 18 items...)
]
```

You control the maximum number of printed elements (or characters) with `string_max` (default: 1024), `array_max` and `hash_max` (both default to 50).

If the amount overflows that limit, you pick which part to show with `string_preserve`, `array_preserve` and `hash_preserve`. They all default to `begin`, meaning it will show the first elements/chars, but you can set them to `end`, `middle`, `extremes` and `none`.

Finally, the overflow message "*(...skipping X items...)*" is set via `string_overflow`, `array_overflow` and `hash_overflow`, and you can use *__SKIP__* as a placeholder that will be replaced with the amount of characters/elements actually skipped.

Hope this helps! And thanks again for taking the time to write this issue and follow-up on it :)

**Comment by GianniGi** [comment]
Great. Thank you very much, I'm happy for that and I'll try ASAP

Il lunedì 14 maggio 2018, Breno G. de Oliveira <notifications@github.com>
ha scritto:
